src/Orleans/Core/Grain.cs

```diff
@@ -39,7 +36,7 @@ protected IGrainFactory GrainFactory
         /// </summary>
         protected IServiceProvider ServiceProvider 
         {
-            get { return Runtime?.ServiceProvider; }
+            get { return Data.ServiceProvider ?? Runtime?.ServiceProvider; }
```

Comment:
> until we provide the option to avoid scopes

I think we should avoid that option for two reasons:
1. Simplicity
2. Allows us to provide services (eg, facets) which are scoped to an activation